src/app/home/home.component.html

```diff
@@ -0,0 +1,119 @@
+                ID Token
+              </button>
+            </h2>
+            <div id="flush-collapseThree" class="accordion-collapse collapse" aria-labelledby="flush-headingThree"
+              data-bs-parent="#accordionSection">
+              <div class="accordion-body">
+                <div class="col">
+                  <code>
+                    {{tokenObjectForDisplay['idToken'] }}
+                  </code>
+                </div>
+                <button type="button" class="btn btn-primary"
+                  (click)="copyToClipboard(tokenObjectForDisplay['idToken'])">Copy</button>
+              </div>
+            </div>
+          </div>
+          <div class="accordion-item">
+            <h2 class="accordion-header" id="flush-headingFour">
+              <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+                data-bs-target="#flush-collapseFour" aria-expanded="false" aria-controls="flush-collapseFour">
+                ID Token Claims
+              </button>
+            </h2>
+            <div id="flush-collapseFour" class="accordion-collapse collapse" aria-labelledby="flush-headingFour"
+              data-bs-parent="#accordionSection">
+              <div class="accordion-body">
+                <code>
+                  <div *ngFor="let key of tokenObjectForDisplay['claimsKeys']">
+                    {{key}}: {{tokenObjectForDisplay['claims'][key]}}
+                  </div>
+                </code>
+              </div>
+            </div>
+          </div>
+          <div class="accordion-item"
+            *ngIf="tokenObjectForDisplay['refreshToken'] && tokenObjectForDisplay['refreshToken'].length > 0">
+            <h2 class="accordion-header" id="flush-headingFive">
+              <button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+                data-bs-target="#flush-collapseFive" aria-expanded="false" aria-controls="flush-collapseFive">
+                Refresh Token
+              </button>
+            </h2>
+            <div id="flush-collapseFive" class="accordion-collapse collapse" aria-labelledby="flush-headingFive"
+              data-bs-parent="#accordionSection">
+              <div class="accordion-body">
+                <code>
+                  {{tokenObjectForDisplay['refreshToken'] }}
+                </code>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+    </div>
+  </ng-container>
+
+  <ng-container *ngIf="!returnTokenStatus()">
+    <div class="row row-padding ">
+      <span class="center">No valid tokens found. Please login, or check your IDP for issues with token
+        generation.</span>
+    </div>
+    <div class="row row-padding">
+      <div class="col center">
+        <button type="button" class="btn btn-primary" (click)="login()">Login (Auth Code Flow)</button>
+      </div>
+    </div>
+  </ng-container>
```

Comment:
Best practice is to add an additional blank line at the end of the file (Unix file conventions)